### PR TITLE
Fix sticky sidebar issues

### LINKED ISSUE #601: [content]: Sidebar is not scrollable
**Is this a BUG REPORT or FEATURE REQUEST?**:
bug

**What happened**:
When sidebar is sticky and there is ToC fragment in it, it's not scrollable.

**What you expected to happen**:
The sidebar should be scrollable even if it's sticky.

**How to reproduce it (as minimally and precisely as possible)**:
Enable sticky option in sidebar config and use ToC inside it.

**Anything else we need to know?**:

**Environment**:
- Syna Theme version:  `0.15.0`
- Hugo version:  `Hugo Static Site Generator v0.55.0-4333CC77F/extended linux/amd64 BuildDate: 2019-04-08T17:05:23Z`
- Others:

### LINKED ISSUE #700: Sidebar with sticky = true
**Is this a BUG REPORT or FEATURE REQUEST?**:
bug

**What happened**:
The sidebar sticky = true doesn't stick !

> content.html (**content** fragment)

if you choose `sticky = true` 
for the sidebar in a content fragment. The sidebar doesn't stick.

**What you expected to happen**:
the side should stick when one scroll down.
but it doesn't !

**How to reproduce it (as minimally and precisely as possible)**:
To reproduce get a content fragment with a sidebar (either left or right)
set the sticky Boolean to true, check in the source code that the class 

> sticky-top pt-3

is set for the child element of `content-sidebar`
Then scroll down, you'll see that the element is not sticky.

**Anything else we need to know?**:
I tried with 

- Brave Version 1.2.43 Chromium: 79.0.3945.130 (Official Build) unknown (64-bit)
- and Firefox 72.0.2 (64-bit). 

I think that the problem comes from the fact that the parent element of #sticky-top  div :
aka : 
```
.content-sidebar {
  word-break: break-word;
  overflow-y: auto;
}
```
has got the style :   **overflow-y: auto;**

I made it work by adding one div around the first _sticky-top_ :

```
        {{- if .sticky }}
          <div class="sticky-top pt-3"> <div class="pt-3">
        {{- end -}}
```
[...]

```
    {{- if .sticky }}
          </div></div> {{/* .sticky-top */}}
          <div></div>
        {{- end -}}
```
and I add in _syna.css:

```
.sticky-top{
    top: 50px;
}
```
and remove **overflow-y** from :

```
.content-sidebar {
  word-break: break-word;
/*   overflow-y: auto; */
}
```

**Environment**:
- Syna Theme version:  v0.17.0
- Hugo version:  Hugo Static Site Generator v0.63.2/extended linux/amd64 BuildDate: unknown
- Others: Arch linux

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
- Fix sticky sidebar scroll issue (Content was not scrollable when it exceeded height of the page)
- Fix sticky sidebar being overlapped by sticky navbar

**Which issue this PR fixes**:
fixes #700 
fixes #601 

**Release note**:
```release-note
- content: Fix sticky sidebar issues
```

### CODE REVIEW

**Comment by mpourismaiel** [comment]
I forgot to mention, please scroll the docs page with some of the articles expanded, in a long page with browser height set to 400px for example. The sidebar should stay in place while you scroll until it meets the header.

The code is implemented in a way that the two fragments stay independent and ignorant of each other. The content fragment only applies the scroll code if there is an active sidebar and should not break if there is a navbar or not.

**Comment by mpourismaiel** [comment]
Fixed the bug that was causing the failing test and then fixed a faulty test that failed occasionally. It was failing for this PR so pushed the commit here, let me know if you want me to separate it.

File: layouts/partials/fragments/nav.html [hunk 1]
```diff
@@ -224,3 +224,10 @@
   });
 </script>
 {{- end -}}
+{{- if .Params.sticky }}
+<style>
+  :root {
+    --header-height: 56px;
+  }
+</style>
+{{- end -}}
```
Comment (stp-ip) [inline layouts/partials/fragments/nav.html:233]: Not sure I like the hardcoding of that. Can't think of a better solution so.

File: layouts/partials/fragments/content.html [hunk 2]
```diff
@@ -62,3 +62,28 @@ <h3 class="sidebar-title {{- printf " text-%s" $title_align -}}">
...
+      var extraPadding = 0;
+
+      if (sidebarTop <= headerHeight) {
+        extraPadding = headerHeight;
+      }
+
+      sidebarContentParent.style.setProperty('padding-top', extraPadding + sidebarContentParentPadding + 'px', 'important');
+    });
+  })();
+</script>
```
Comment (stp-ip) [inline layouts/partials/fragments/content.html:89]: Why inline JS instead of within our JS scripts?

**Comment by stp-ip** [comment]
Like the scrolling, but the position doesn't seem perfect. This is more a side effect of #513 I guess. As we are overwriting the height. Would that be something we would like to fix here as well or am I missing something?

**Comment by stp-ip** [comment]
Edgecase, that should probably get a dev/ page and maybe a test. What happens with 2 sidebars? Do we need an additional note for the "sticky" config for the content fragment?

File: layouts/partials/fragments/nav.html (hunk 1 above)
Comment (mpourismaiel) [reply layouts/partials/fragments/nav.html:233]: I can't think of a solution that doesn't involve JS and is not hard coding the value. I'll try to figure something out and maybe we can create an issue to track it.

Comment (stp-ip) [reply layouts/partials/fragments/nav.html:233]: Fine to keep it as is and open a "revisit" to track this.

**Comment by stp-ip** [comment]
What do you think about a dev/ page for manual testing and probably also automated testing?

**Comment by mpourismaiel** [comment]
@stp-ip Added tests and dev pages accordingly.

**Pull request merged**